Tools/AP_Periph/can.cpp

```diff
@@ -1559,7 +1559,10 @@ void AP_Periph_FW::can_battery_update(void)
         }
 
         pkt.state_of_health_pct = UAVCAN_EQUIPMENT_POWER_BATTERYINFO_STATE_OF_HEALTH_UNKNOWN;
-        pkt.state_of_charge_pct = battery.lib.capacity_remaining_pct(i);
+        int8_t percentage = 0;
```

Comment:
We've moved from returning `uint8_t` to filling in an `int8_t`.  That's rather odd for the battery library IMO - a battery shouldn't ever have a negative remaining percent.  I realise that some transports have special meanings for `-1` and whatnot, but internally this should probably still have been unsigned.

I'm not that fussed by this change, however, so don't feel the need to change this.